### PR TITLE
Add new resource to enable logging to seq

### PR DESCRIPTION
**Pull request by matt-richardson**

### CODE REVIEW

File: OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1 [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+    $nlogTargetElement = ($nlogConfig.nlog.targets.target | where-object {$_.name -eq "seq"})
+    $nlogTargetElementExists = $null -ne $nlogTargetElement
+    $nlogRuleElement = ($nlogConfig.nlog.rules.logger | where-object {$_.writeTo -eq "seq"})
+    $nlogRuleElementExists = $null -ne $nlogRuleElement
+
+    $plainTextPassword = ($nlogConfig.nlog.targets.target | where-object { $_.name -eq "seq" -and $_.type -eq "Seq" }).ApiKey
+    if ($null -ne $plainTextPassword) {
+      $password = new-object securestring
+      # Avoid using "ConvertTo-SeureString ... -AsPlaintext", to fix PSAvoidUsingConvertToSecureStringWithPlainText
+      # Not sure it actually solves the underlying issue though
```
Comment (stopthatastronaut) [inline OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1]: I'm not entirely comfortable about this bit either. but it does get round the script analyzer...

File: OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1 [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+    $nlogExtensionElement = ($nlogConfig.nlog.extensions.add | where-object { $_.assembly -eq "Seq.Client.Nlog" })
+    if ($null -ne $nlogPropertyElement) {
+      $nlogConfig.nlog.extensions.RemoveChild($nlogExtensionElement)
+    }
+    $newChild = $nlogConfig.CreateElement("add", $nlogConfig.DocumentElement.NamespaceURI)
+    $newChild.Attributes.Append((New-XmlAttribute $nlogConfig "assembly" "Seq.Client.NLog"))
+    $nlogConfig.nlog.extensions.AppendChild($newChild)
+
+    #remove then re-add "
+    #  <target name="seq" xsi:type="Seq" serverUrl="https://seq.octopushq.com" apiKey="wGdZV8CUL92KsGqZNVDX">
```
Comment (stopthatastronaut) [inline OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1]: Is this a valid API key?

File: OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1 [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+    Write-Verbose "(FOUND MISMATCH) Configuration parameter '$key' with value '********' mismatched the specified value '********'"
+    return $false
+  }
+  else {
+    Write-Verbose "Configuration parameter '$key' matches the requested value '********'"
+  }
+  return $true
+}
+
+function Get-OctopusDSCParameter($parameters) {
```
Comment (stopthatastronaut) [inline OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1]: I feel like we should have a shared code module for bits like this. I'll open a separate isue for that.

File: OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1 (hunk 2 above)
Comment (matt-richardson) [reply OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1]: Whoops. It was, yeah. Disabled on the seq side.

File: OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1 (hunk 3 above)
Comment (matt-richardson) [reply OctopusDSC/DSCResources/cOctopusSeqLogger/cOctopusSeqLogger.psm1]: We should, yeah. Lots of common code. Even the implementation of `Test-TargetResource` is the same across most resources

**Pull request merged**